Automattic/wp-calypso
# Odyssey Stats: use WPCOM site info as defaults

This is not the primary purpose, but it fixes #81538 as well. 

## Proposed Changes

[is_commercial is a WPCOM option](https://github.com/Automattic/jetpack/pull/32595), which we need to query  from WPCOM.As some of the information might not be accurate, we are going to use them as defaults with the options in initial state from Jetpack as source of truth.

The PR proposes to add the site info querying in the process of `siteSelection` middleware in `apps/odyssey-stats/src/routes.ts`, where queries `jetpack/v4/site`, decodes the data, and dispatches Redux actions to bridge the whole data flow.

As the old reducer would override the whole site info with data from WPCOM, the PR has to add a new reducer for action `ODYSSEY_SITE_RECEIVE` which uses site info from wpcom as defaults and overrides it with values from initial state.

## Testing Instructions

* Build Jetpack if needed `jetpack build plugins/jetpack`
* Add a line at: 
https://github.com/Automattic/wp-calypso/blob/f3bf9699554234847c856172c7dfe1511fbc1ab6/apps/odyssey-stats/src/app.tsx#L59
```
setInterval(()=> console.log(store.getState()), 5000);// Or anyone can remind me the  middleware to attach the redux state to window? couldn't find it anywhere anymore 🤷
```
* Build Odyssey Stats `cd apps/odyssey-stats && STATS_PACKAGE_PATH=~/path/to/jetpack/projects/packages/stats-admin yarn dev`
* Open `/wp-admin/admin.php?page=stats#!/stats/day/:siteSlug`
* Check console
* Ensure you could find `is_commercial` in sites -> items -> :blog_id -> options

**Note: the buttons to purchase Stats do not work at the moment and we'll fix those in follow up PRs.**



## Pre-merge Checklist
